File: java/MaxNumPrice.java
package algo;

import java.util.*;

public class MaxNumPrice {

	private static int naive(int n) {
		long sum = 0;
		int k = 1;
		
		for (int i = 1 ; i < n + 1 ; i++) {
			sum += i;
			if (sum > n) {
				k = i - 1;
				break;
			}
		}
		
		return k;
	}
	
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		
		int n = scanner.nextInt();
		System.out.println(naive(n));
		
		scanner.close();
	}
}

File: python/MaxNumPrice.py

def naive(n):
  k = 1
  sum = 0
  
  for i in range(1, n + 1):
    sum += i
    if sum > n:
      k = i - 1
      break

  return k

n = int(input())
k = naive(n)
print(k)
for i in range(1, k + 1):
  if i == k:
    print(i + n - (i*(i+1)//2), end = '')
  else:
    print(i, end = ' ')
